realm/realm-library/src/syncIntegrationTest/java/io/realm/SyncedRealmTests.java

```diff
@@ -179,6 +178,7 @@ public void onError(Throwable exception) {
             SystemClock.sleep(randomizer.nextInt(5));
             task.cancel();
         }
+        TestHelper.waitRealmThreadExecutorFinish();
```

Comment:
The RunTestInLooperThread does this automatically: https://github.com/realm/realm-java/blob/master/realm/realm-library/src/androidTest/java/io/realm/rule/RunInLooperThread.java#L287 ?

Reply:
hmmm, right. i missed that. then it probably has the same issue with https://github.com/realm/realm-sync/issues/1770